packages/core/core-flows/src/order/workflows/exchange/refresh-shipping.ts

```diff
@@ -0,0 +1,150 @@
+        "actions.*",
+      ],
+      variables: {
+        filters: {
+          order_id: input.order_id,
+          exchange_id: input.exchange_id,
+          status: [OrderChangeStatus.PENDING, OrderChangeStatus.REQUESTED],
+        },
+      },
+      list: false,
+    }).config({ name: "order-change-query" })
+
+    const refreshArgs = transform(
+      { input, orderChange },
+      ({ input, orderChange }) => {
+        const result: Record<string, any> = []
+
+        const inboundShippingAction = orderChange.actions.find(
+          (action) =>
+            action.action === ChangeActionType.SHIPPING_ADD &&
+            !!action.return_id
+        )
+
+        const outboundShippingAction = orderChange.actions.find(
+          (action) =>
+            action.action === ChangeActionType.SHIPPING_ADD && !action.return_id
+        )
+
+        if (inboundShippingAction) {
+          const items = orderChange.actions
+            .filter((action) => action.action === ChangeActionType.RETURN_ITEM)
+            .map((a) => ({
+              id: a.details?.reference_id as string,
+              quantity: a.details?.quantity as number,
+            }))
+
+          result.push({
+            shipping_method_id: inboundShippingAction.reference_id,
+            order_id: orderChange.order_id,
+            action_id: inboundShippingAction.id,
+            context: {
+              return_id: inboundShippingAction.return_id,
+              return_items: items,
+            },
+          })
+        } else {
+          result.push(null)
+        }
+
+        if (outboundShippingAction) {
+          const items = orderChange.actions
+            .filter((action) => action.action === ChangeActionType.ITEM_ADD)
+            .map((a) => ({
+              id: a.details?.reference_id as string,
+              quantity: a.details?.quantity as number,
+            }))
+
+          result.push({
+            shipping_method_id: outboundShippingAction.reference_id,
+            order_id: orderChange.order_id,
+            action_id: outboundShippingAction.id,
+            context: {
+              exchange_id: outboundShippingAction.exchange_id,
+              exchange_items: items,
+            },
+          })
+        } else {
+          result.push(null)
+        }
+
+        return result
+      }
+    )
+
+    // Refresh inbound shipping method
+    when({ refreshArgs }, ({ refreshArgs }) => refreshArgs[0] !== null).then(
+      () =>
+        maybeRefreshShippingMethodsWorkflow
+          .runAsStep({
+            input: refreshArgs[0],
+          })
+          .config({ name: "refresh-inbound-shipping-method" })
+    )
+
+    // Refresh outbound shipping method
+    when({ refreshArgs }, ({ refreshArgs }) => refreshArgs[1] !== null).then(
+      () =>
+        maybeRefreshShippingMethodsWorkflow
+          .runAsStep({
+            input: refreshArgs[1],
+          })
+          .config({ name: "refresh-outbound-shipping-method" })
+    )
```

Comment:
thought: What happens if there are more than one outbound or inbound shipping method on the exchange?

Reply:
Fine to keep for a follow up iteration.